### PR TITLE
Filter first and last log files on timestamp, and don't quit reading …

### PR DESCRIPTION
**Pull request by jonmv**
…too early

@freva please review.

### CODE REVIEW

**Comment by jonmv** [comment]
Say three log files contains three entries each, logged at times 1, 2, 3; 4, 5, 6; and 7, 8, 9. Their modification timestamps are 3, 6, 9. 

Assume we ask for everything between 5-delta and 5+delta. Previously, the second file would not be considered, because 6 isAfter 5+delta, even though it contains two entries of interest. 

This PR fixes that. 

This PR also filters out entries in the first and last log files, ensuring the given time window is honoured precisely.

**Comment by jonmv** [comment]
Lots of pointless information in that example, but you get it :p

**Review by freva — APPROVED** [review]
Great! :100: :1st_place_medal: :fire: 

Finally we can remove this hack:
https://git.ouroath.com/vespa/dashboard-app/blob/master/app/components/apps/instance-log-panel.js#L37

File: container-core/src/main/java/com/yahoo/container/handler/LogReader.java [hunk 1]
```diff
@@ -76,13 +127,20 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) {
...
-        return paths.stream()
-                .sorted(Comparator.comparing(Pair::getSecond))
-                .map(Pair::getFirst)
-                .collect(Collectors.toList());
+        List<Path> sorted = new ArrayList<>();
+        for (var entries = paths.entrySet().stream().sorted(comparing(Map.Entry::getValue)).iterator(); entries.hasNext(); ) {
+            var entry = entries.next();
+            if (entry.getValue().isAfter(from))
+                sorted.add(entry.getKey());
+            if (entry.getValue().isAfter(to))
```
[the remark this answers is not in the document]
Comment (jonmv) [reply container-core/src/main/java/com/yahoo/container/handler/LogReader.java]: o_O

Well, would have to use a boolean reference, as the first item where the predicate fails, should also be included. But ... O_o

[the remark this answers is not in the document]
Comment (freva) [reply container-core/src/main/java/com/yahoo/container/handler/LogReader.java]: It is not, f.ex.:
```
IntStream.range(0, 1000)
        .dropWhile(i -> i < 544)
        .takeWhile(i -> i < 549)
        .forEach(System.out::println);
```
Gives:
```
544
545
546
547
548
```

[the remark this answers is not in the document]
Comment (jonmv) [reply container-core/src/main/java/com/yahoo/container/handler/LogReader.java]: Exactly, but I want it to be.

**Comment by freva** [comment]
Ah yes, I can't read :man_facepalming:

File: container-core/src/main/java/com/yahoo/container/handler/LogReader.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (freva) [reply container-core/src/main/java/com/yahoo/container/handler/LogReader.java]: Ah yes, I can't read :man_facepalming:

**Pull request merged**